test/infrastructure/vcsim/README.md

```diff
@@ -1,4 +1,156 @@
 # vcsim controller
 
-vcsim controller provides one or more vcsim instances, as well as the fake API server / etcd running on the
-simulated machines.
+vcsim controller is a controller that provides one or more vcsim instances, as well as the VIP for control
+plane endpoints.
+
+## Architecture
+
+vcsim controller is a regular Kubernetes controller designed to be run aside to CAPV when you are planning to use vcsim
+as a target infrastructure instead of a real vCenter.
+
+It is also worth to notice that vcsim controller leverage several components from Cluster API's in-memory
+provider, and thus it is recommended to become familiar with this [document](https://github.com/kubernetes-sigs/cluster-api/blob/main/test/infrastructure/inmemory/README.md)
+before reading following paragraphs.
+
+### Preparing up a test environment (using vcsim)
+
+In order to understand the architecture of the vcsim controller, it is convenient to start looking only at 
+components that are involved in setting up a test environment that will use vcsim as a target infrastructure.
+
+A test environment for CAPV requires two main elements:
+
+- A vCenter, or in this case a vcsim instance simulating a vCenter.
+- A VIP for the control plane endpoint
+
+In order to create a vcsim instance it is possible to use the `VCenterSimulator` resource, and the corresponding
+VCenterSimulatorReconciler will take care of the provisioning process.
+
+Note: The vcsim instance will run inside the Pod that host the vcsim controller, and be accessible trough a port that
+will surface in the status of the `VCenterSimulator` resource.
+
+Note: As of today, given a limitation in the vcsim library, only a single instance vcsim instance can be active at any
+time.
+
+In order to get a VIP for the control plane endpoint it is possible to use the `ControlPlaneEndpoint` resource, 
+and the corresponding ControlPlaneEndpointReconciler will take care of the provisioning process.
+
+Note: The code for CAPI's in memory provider is used to create VIPs; they ar implemented as a listener on a port
+in the range 20000-24000 on the vcsim controller Pod. The port will surface in the status of the `ControlPlaneEndpoint`
+resource.
+
+![Architecture](architecture-part1.drawio.svg)
```

Comment:
Is this picture correct? It shows that the InMemoryClusterReconciler creates a VIP based on InMemoryClusters? I would have expected something with ControlPlaneEndpoint CRDs